contribs/sharing/src/main/java/org/matsim/contrib/sharing/logic/SharingLogic.java

```diff
@@ -122,6 +129,10 @@ public boolean tryBookVehicle(double now, MobsimAgent agent) {
 			// Insert new plan elements
 			plan.getPlanElements().addAll(bookingActivityIndex + 1, updatedElements);
 
+			Leg leg = ((Leg) mainElements.get(0));
```

Comment:
This is brittle, there can be more stuff than a leg. Maybe there is some ideas in the TripRouter on how to cleanly set the vehicle for a multi-stage trip. I guess "car with access and egress walk" is not the best example here, because it will probably not happen (except we can define a station as some "virtual" thing and the vehicles are parked in the links around that station). But since we want to be generic there could be all kind of things like "a car trip with a refueling activity in the middle" or whatever :)

Reply:
Implemented setVehicle(Plan plan, List<? extends PlanElement> planElements, SharingVehicle vehicle) with IllegalStateException if more than 1 vehicleId was present